### DISCUSSION TITLE
git error fix by updates today...

### DISCUSSION THREAD

**Discussion by PowerfulShell** [vladmandic/sdnext#408] [Q&A]
git errors magically resolved from update this repo today, whatever

**Comment by SGKino** [comment]
Same here. With the M2 environment.
It seems that the system can't find a proper version of tensorflow.

**Comment by SGKino** [comment]
Got the same error, even I have upgraded to 2.12 mannually.
<img width="704" alt="Screenshot 2023-04-22 at 17 01 06" src="https://user-images.githubusercontent.com/27273809/233774581-1d181765-2f68-4e50-8342-2776a54f9316.png">

**Comment by dilectiogames** [comment]
anyone knows what is this thing and how to make it work

**Comment by vladmandic** [comment]
@SGKino this issue is not about M2 or Tensorflow

@bryanray those are not small errors this is basically "none of git ops are working", there is something broken in the way your " self contain system folder for python an git, they are all portable" is setup - and i have no idea what this means?
also, check actual `setup.log` for more details.

**Comment by SGKino** [comment]
> @SGKino this issue is not about M2 or Tensorflow
> 
I know.
Just want to report my tensoflow issue.

That is the only error I have got.
I got a very slow performance speed, around 40+s latency during the upscaling.
Wonder if this due to the tensoflow.

**Comment by vladmandic** [comment]
looks like version of `git` in your portable installation is too old and doesn't understand all necessary commands:
> git: 'submodule' is not a git command. See 'git --help'.

what is your git version (`git --version`)? i'm using 2.34, but command `submodule` was added a while back, so your git must be pretty old.

**Comment by vladmandic** [comment]
> still no luck :(

what is "no luck"?
if you run `git submodule` in repo root, what does it say?

**Comment by vladmandic** [comment]
run it in root folder where you cloned the repo, not in git folder itself.

**Comment by vladmandic** [comment]
you set path to git in your batch. set it manually as well so you can run git in command prompt yourself. this path issue is very basic.

**Comment by vladmandic** [comment]
i'm trying to help you and you're not making it easy.
your batch seems fine, the issue is that your `git` did not recognize submodules command at all. now you've updated `git` and we need to test if now it does - and for that, you need to be able to run git yourself - which you seem either reluctant or unable to do. why it works in a1111 and not here? a1111 does not utilize submodules, as simple as that.

anyhow, moving this to discussions as this is not a repo issue, this is python/git toolkit issue.

**Comment by dilectiogames** [comment]
why you keep moving issues without resolving them? There is clearly an issue with how your setup is installing stuff.

git obviously works, I've installed git and downloaded both your app and the stable diffusion app. That one worked fine using same git installation.

I have the same issues and I have no portable git anywhere.

**Reply by vladmandic** [reply]
@bryanray set PATH using same line as in your batch file.
I cannot copy&paste it since you did not provide text, but only screenshots.
then run git submodule as i've asked.

**Comment by vladmandic** [comment]
> why you keep moving issues without resolving them? There is clearly an issue with how your setup is installing stuff.

Clearly? To me this is clearly issue with portable git not understanding submodule command at all. Not issue with repo.
And submitter could not even run it manually after me attempting for a day. And all i asked is simple add-to-path and check.

> git obviously works, I've installed git and downloaded both your app and the stable diffusion app. That one worked fine using same git installation.

I require newer git that understands submodule command. As simple as that

> I have the same issues and I have no portable git anywhere.

You do? Then say so. Don't complain. Don't say "me too" without providing ANY info. You're not being helpful. And I'm trying as much as possible to help everyone. But your attitude does not help.

**Reply by vladmandic** [reply]
> please help, i already posted batch

yes, you did.
and its sunday. and i worked on OPEN SOURCE code for 60+h this week. and this is not even repo issue, its issue with external tools.  
i'll do my best, but you're really not motivating.

**Reply by dilectiogames** [reply]
> > why you keep moving issues without resolving them? There is clearly an issue with how your setup is installing stuff.
> 
> Clearly? To me this is clearly issue with portable git not understanding submodule command at all. Not issue with repo. And submitter could not even run it manually after me attempting for a day. And all i asked is simple add-to-path and check.
> 
> > git obviously works, I've installed git and downloaded both your app and the stable diffusion app. That one worked fine using same git installation.
> 
> I require newer git that understands submodule command. As simple as that
> 
> > I have the same issues and I have no portable git anywhere.
> 
> You do? Then say so. Don't complain. Don't say "me too" without providing ANY info. You're not being helpful. And I'm trying as much as possible to help everyone. But your attitude does not help.

I've said that but you must have not seen the comment.

sorry for appearing aggressive, keep up the good work. :)
